File: 12.recursive_descent/rd.py
def match(ss):
	global i
	global look
	i=i+1
	look=s[i]

def E():
	global flag
	global look
	if look=="n":
		match("n")
		T()
	if look=="$":
		flag=1
		print "SUCCESS"
def T():
	global d
	global look
	a=d['T']
	if look=="*":
		match("*")
		if look=="n":
			match("n")
			T()
		else:
			print "Rejected"
			exit()
	else:
		for i in range(len(a)):
			if "{" in a[i]:
				return
		print "Rejected"
		exit()

flag=0
d=dict()
d['E']=["nT$"]
d['T']=["*nT","{"]
print d
s=raw_input("Enter the string")
s=s+"$"
i=0
look=s[i]
E()
if flag==0:
	print "Rejected"

File: 12.recursive_descent/rd.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;

void E(char, string[], string);
void T(string[], string, int);
int main()
{
	string productions[]={"E->nT$", "T->*nT", "T->#"};
	char start_state='E';
	string input;
	cout<<"Enter string to check\n";
	cin>>input;

	E(start_state, productions, input);
}

void E(char s, string productions[], string input)
{
	static int j=0;
	int i=3;
	char lookhead=input[j++];
	if(lookhead==productions[0][i++] && j<=input.length())
	{
		T(productions, input, j);	
	}
	else
		cout<<"Rejected\n";
	if(input[j]=='$')
		cout<<"Success\n";
}

void T(string productions[], string input, int loc)
{
	int i=3;
	char lookhead=input[loc++];
	
	if(lookhead==productions[1][i++] && loc<input.length())
	{
		if(input[loc++]==productions[1][i++])
		{
			lookhead=input[loc-1];
			T(productions, input, loc);
			if(input[loc]=='$')
				cout<<"Success\n";
			else
				cout<<"Rejected\n";
		}
		else
			cout<<"Rejected";
	}
	return;
}
